skietbaan_fe/src/components/LeaderboardPage.js

```diff
@@ -1,143 +1,325 @@
+        this.getLeaderboardData();
+    }
+    getLeaderboardData() {
+        let token = getCookie("token"); 
+        console.log("displaying tokken")
+        console.log(token);
+        const filterSelection = {
+            selectedCompetition: this.state.selectedGroup,
+            selectedGroups: this.state.selectedCompetition,
+            selectedScoreType: this.state.selectedScoreType,
+            userToken: token
         }
+        this.props.fetchleaderboadtabledata(filterSelection);
+        console.log("filter selection");
+        console.log(filterSelection);
+    }
 
-
-    setCompetitionValue = (event) => {
+    setCompetitionValue = (value) => {
         this.setState({
-            selectedCompetition: event.value
+            selectedCompetition: value
         })
-        this.updateLeaderboard();
+        this.getLeaderboardData();
     }
-    setGroupValue = (event) => {
+    setGroupValue = (value) => {
         this.setState({
-            selectedGroup: event.value
+            selectedGroup: value
         });
-        this.updateLeaderboard();
+        this.getLeaderboardData();
     }
-    setScoreTypeValue = (event) => {
+    setScoreTypeValue = (value) => {
         this.setState({
-            selectedST: event.value
+            selectedScoreType: value
         });
-        this.updateLeaderboard();
+        //this.getLeaderboardData();
     }
+    displayScoreByType = (result) => {
 
-    updateLeaderboard() {
-        fetch('http://localhost:63474/api/Leaderboard/GetLeaderboardRankings?competitionID=' + this.state.selectedCompetition + '&groupID=' + this.state.selectedGroup + '&ScoreType=' + this.state.selectedST)
-            .then(res => res.json())
-            .then(data => this.setState({ rankingTable: data }, console.log(data), console.log(this.state.rankingTable)));
-            console.log("State Has been updated in updateLeaderboard()")
+        if (this.state.selectedScoreType == 0) {//average
+            return result.average;
+        } else if (this.state.selectedScoreType == 1) {//total
+            return result.total;
+        } else if (this.state.selectedScoreType == 2) {//best
+            return result.bestScore;
+        }
     }
+    getCurentUserRankNumber(rankingList) {
+        console.log(rankingList);
+        if (rankingList != undefined) {
+            for (var i = 0; i < rankingList.length; i++) {
+                if (rankingList[i].username === this.props.userResults.username) {
+                    return rankingList[i].rank;
+                }
+            }
+            return null;
+        } else {
+            return null;
+        }
 
 
-
+    }
+    showMoreScores(index) {
+        console.log("index -->");
+        console.log(index);
+        if (this.props.tableData[index].showMore) {
+            this.props.tableData[index].showMore = false;
+        } else {
+            this.props.tableData[index].showMore = true;
```

Comment:
remove else and just say this.props.tableData[index].showMore = true -> should also be in the if statement that checks that this array exists